platform/mbed_rtc_time.cpp

```diff
@@ -76,13 +76,31 @@ static void (*_rtc_write)(time_t t) = NULL;
 #ifdef __cplusplus
 extern "C" {
 #endif
-#if defined (__ICCARM__)
-time_t __time32(time_t *timer)
-#else
-time_t time(time_t *timer)
-#endif
 
+int settimeofday(const struct timeval *tv, MBED_UNUSED const void *tz)
 {
+    if (tv == NULL) {
+        return -1;
+    }
+
+    _mutex->lock();
+    if (_rtc_init != NULL) {
+        _rtc_init();
+    }
+    if (_rtc_write != NULL) {
+        _rtc_write(tv->tv_sec);
+    }
+    _mutex->unlock();
+
+    return 0;
+}
+
+int gettimeofday(struct timeval *tv, MBED_UNUSED void *tz)
+{
+    if (tv == NULL) {
```

Comment:
I personally would not bother with a null check on API like this or `settimeofday`. We're not made of ROM, and nobody's going to be checking the return code. (I know others like them though).